modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuildController.java

```diff
@@ -180,14 +180,19 @@ private CompletableFuture<?> onPrimaryReplicaElected(PrimaryReplicaEventParamete
                 // metastore thread.
                 int catalogVersion = catalogService.latestCatalogVersion();
 
-                return getMvTableStorageFuture(parameters.causalityToken(), primaryReplicaId)
-                        .thenCompose(mvTableStorage -> awaitPrimaryReplica(primaryReplicaId, parameters.startTime())
-                                .thenAccept(replicaMeta -> tryScheduleBuildIndexesForNewPrimaryReplica(
+                CompletableFuture<ReplicaMeta> replicaMetaFuture = awaitPrimaryReplica(primaryReplicaId, parameters.startTime());
+
+                CompletableFuture<MvTableStorage> mvTableStorageFuture =
+                        getMvTableStorageFuture(parameters.causalityToken(), primaryReplicaId);
+
+                return replicaMetaFuture
+                        .thenAcceptBoth(mvTableStorageFuture, (replicaMeta, mvTableStorage) ->
```

Comment:
Why do you think that we need to wait for these two futures in parallel?
Let's say `mvTableStorageFuture` returns `null`, then it will be NPE, how do you think this will fix the problem from the ticket?

Reply:
reverted, you are right